### PR TITLE
Implemented simulation launcher for network version

### PR DESCRIPTION
**Pull request by netsirius**
### Changes summary
This pull request adds network mode to the test launcher, enabling simulations in a more realistic network environment. Initially uses local TCP sockets, laying the groundwork for future expansion to more generic settings, including cloud environments.

### Key Implementation Details
- **Setup and Initialization**: Completed initial setup of the web server and node configuration, establishing the core framework for network simulation.
- **Node and Server Interaction**: Configured nodes to retrieve peer settings from the server, preparing the server to manage and dispatch events to peers.
- **P2P Node Execution**: Implemented with a new event generator and using the MemoryContractHandler.
- **Connectivity Achievement**: Successfully established connections with nodes. However, an issue has been identified involving the discrepancy between P2P peer IDs and the peer IDs we generate. Due to potential shift from current P2P library, this issue is not prioritized for immediate fix.

### CODE REVIEW

File: crates/core/src/client_events.rs [hunk 1]
```diff
@@ -316,6 +320,113 @@ pub(crate) mod test {
         }
     }
 
+    pub struct NetworkEventGenerator<R = rand::rngs::SmallRng> {
+        id: PeerId,
+        memory_event_generator: MemoryEventsGen<R>,
+        ws_client: Arc<Mutex<WebSocketStream<MaybeTlsStream<TcpStream>>>>,
+        rng: Option<R>,
```
Comment (iduartgomez) [inline crates/core/src/client_events.rs:327]: does this need to be an option if is always set?

File: crates/core/src/ring.rs [hunk 2]
```diff
@@ -660,7 +660,11 @@ impl Ring {
             tracing::info!(%peer, "Removing connection");
         }
         self.live_tx_tracker.prune_transactions_from_peer(&peer);
-        let loc = self.location_for_peer.write().remove(&peer).unwrap();
+        let Some(loc) = self.location_for_peer.write().remove(&peer) else {
+            self.open_connections
```
Comment (iduartgomez) [inline crates/core/src/ring.rs:664]: maybe add acomment here sayign why we do this:
// this case would be when a connection is being open, so peer location hasn't been recorded yet and we can ignore everything below

File: crates/fdev/src/testing/multiple_process.rs [hunk 3]
```diff
@@ -396,13 +396,12 @@ async fn child(
     child_config: &MultiProcessConfig,
 ) -> anyhow::Result<()> {
     let id = child_config.id.expect("id should be set for child process");
-    let data_dir = child_config
+    let _data_dir = child_config
```
Comment (iduartgomez) [inline crates/fdev/src/testing/multiple_process.rs:399]: if we are not setting this, then we don't need it.
how are we setting where data is stored then though?

**Review by iduartgomez — COMMENTED** [review]
Just some minor things, looking good

File: crates/fdev/src/testing/multiple_process.rs (hunk 3 above)
Comment (netsirius) [reply crates/fdev/src/testing/multiple_process.rs:399]: I'm going to remove it. The data_dir variable was used to set this environment variable: std::env::set_var("FREENET_DATA_DIR", data_dir); Now this env var is not used.

File: crates/core/src/client_events.rs (hunk 1 above)
Comment (netsirius) [reply crates/core/src/client_events.rs]: fixed

File: crates/core/src/ring.rs (hunk 2 above)
Comment (netsirius) [reply crates/core/src/ring.rs:665]: added

File: crates/fdev/src/testing/multiple_process.rs (hunk 3 above)
Comment (netsirius) [reply crates/fdev/src/testing/multiple_process.rs]: fixed

**Pull request merged**